src/aggregator/aggregator/follower_flush_mgr.go

```diff
@@ -279,12 +312,36 @@ func (mgr *followerFlushManager) CanLead() bool {
 	return true
 }
 
+// canLeadNotFlushed determines whether the follower can takeover leadership
+// for the window that was not (yet) flushed by leader.
+// This is case is possible when leader encounters a metric at some resolution
+// window for the first time in the shard it owns.
+func (mgr *followerFlushManager) canLeadNotFushed(now time.Time, windowSize time.Duration) bool {
+	windowStartAt := now.Truncate(windowSize)
+	return mgr.openedAt.Before(windowStartAt)
+}
+
 func (mgr *followerFlushManager) canLead(
+	now time.Time,
+	flusherType metricListType,
+	shardID int,
 	flushTimes map[int64]int64,
 	metrics standardFollowerFlusherMetrics,
 ) bool {
 	for windowNanos, lastFlushedNanos := range flushTimes {
 		windowSize := time.Duration(windowNanos)
+		if lastFlushedNanos == 0 {
+			mgr.logger.Info("Encountered a window that was never flushed by leader",
+				zap.Stringer("windowSize", windowSize),
+				zap.Stringer("flusherType", flusherType),
+				zap.Int("shardID", int(shardID)))
+			mgr.metrics.forwarded.windowNeverFlushed.Inc(1)
+
+			if mgr.canLeadNotFushed(now, windowSize) {
```

Comment:
Nit: maybe more consistent with the other code (below) to do:

```
if !mgr.canLeadNotFlushed(...) {
   return false
}
```

Either way though.

Reply:
If it's a never flushed window and `canLeadNotFlushed(...)` is `true` for it, we have to `continue` to skip the further 
check that assumes the window is flushed window. But if it's false then yes, need to return early. So it actually has to be like this:

```
if mgr.canLeadNotFlushed(now, windowSize) {
    continue
} else {
    return false
}
```

Reply:
Ah yeah I see your point, good catch. Do we have test cases around the `return false` case?